docs/blueprints/EP031.rst

```diff
@@ -0,0 +1,312 @@
+
+  1. Adding INT metadata:
+
+    - match:
+
+      - table 0
+
+        - priority: Higher (mef_eline + 100)
+        - match: <in_port=\ **11**,vlan_vid=20,eth_type=0x800,ip_proto=6>  # TCP
+        - instrutions:
+
+          - action: add_int_metadata
+          - action: output port **1** (loop)
+
+        - priority: Higher (mef_eline + 100)
+        - match: <in_port=\ **11**,vlan_vid=20,eth_type=0x800,ip_proto=17>. # UDP
+        - instrutions:
+
+          - action: add_int_metadata
+          - action: output port **1** (loop)
+
+  2. Send Report and pop INT data (traffic is coming from port 2 that's the loop with port 1). Only INT data gets into the loop.
+
+    - match:
+
+      - table 0
+
+        - priority: Higher (mef_eline + 100)
+        - match: <in_port=\ **2**,vlan_vid=20>
+        - instrutions:
+
+          - action: send_report
+          - action go to table 2
+
+
+      - table 2
+
+        - priority: Higher (mef_eline + 100)
+        - match: <in_port=\ **2**,vlan_vid=20>
+        - instrutions:
+
+          - action: pop_int
+          - action [set_queue, pop_vlan, etc.]
+          - action: output port **23** (original port)
+
+
+IV. How to enable INT for EVCs
+==============================
+
+The goal for the **telemetry** app is to enable telemetry for ALL EVCs. However, it must support enabling and disabling telemetry for a single EVC or ALL EVCs. This is the approach:
+
+  1 . The **telemetry** napp will start operating once **mef_eline** is loaded and EVCs and their flows are pushed to the data plane.
+
+  2. **telemetry** will listen for events *kytos/mef_eline.(redeployed_link_(up|down)|deployed)* and *kytos.mef_eline.created* issued by **mef_eline**.
+
+  3. For each EVC identified, **telemetry** will
+    1. use EVC's cookie to get all flow entries created by **flow_manager** IF telemetry is not already enabled.
+    2. push more specific flows as described in Section II. (See Section IX for information on the cookie ID to be used.)
+    3. add a key in the EVC's metadata called "telemetry" with value "enabled". key "telemetry" will be "disabled" once telemetry is disabled for an EVC.
+
+V. Events
+==========
+
+  1. Listening
+    1. *kytos/mef_eline.(removed|deployed)*
+
+  2. Issuing
+    1.  *kytos.telemetry.enabled*
+    2.  *kytos.telemetry.disabled*
+
+
+VI. REST API
+=============
+
+  - POST /telemetry/v1/evc/ body evc_ids: [] for bulk insertions, if empty, then enable all. If invalid or non-existing EVC_ID are provided, abort the entire operation with 4XX status code.
+  - POST /telemetry/v1/evc/<evc_id>: enable/create INT flows for an EVC_ID.
+  - DELETE /telemetry/v1/evc/ body evc_ids: [] for bulk removals, if empty, then remove all. If invalid or non-existing EVC_ID are provided, abort the entire operation with 4XX status code.
+  - DELETE /telemetry/v1/evc/<evc_id>: disable/remove INT flows for an EVC_ID.
+  - GET /telemetry/v1/evc list all INT-enabled EVCs.
```

Comment:
Currently, `mef_eline` doesn't provide bulk updates for metadata or metadata filtering. We'll need to map new requirements on `mef_eline` unless we change this approach and model a new collection for storing INT state.

Reply:
Captured the discussion here https://github.com/kytos-ng/mef_eline/issues/225